frontend/plan/src/syncutils.js

```diff
@@ -119,6 +130,13 @@ const initiateSync = (store) => {
             intervalRecord.push(window.setInterval(syncLoop, SYNC_INTERVAL));
         }));
 
+    window.addEventListener("beforeunload", (e) => {
+        if (!allPushed(store.getState().schedule)) {
+            e.preventDefault();
+            e.returnValue = "";
```

Comment:
Do we want to put a message here? Not exactly sure how this event listener works